opencensus-shim/src/test/java/io/opentelemetry/opencensusshim/OpenTelemetryMetricExporterTest.java

```diff
@@ -0,0 +1,227 @@
+    WaitingMetricExporter exporter = new WaitingMetricExporter();
+    OpenTelemetryMetricsExporter otelExporter =
+        OpenTelemetryMetricsExporter.createAndRegister(exporter, Duration.create(0, 5000));
+    try {
+      TagContext tagContext =
+          tagger
+              .emptyBuilder()
+              .put(tagKey, tagValue, TagMetadata.create(TagMetadata.TagTtl.UNLIMITED_PROPAGATION))
+              .build();
+      try (io.opencensus.common.Scope ss = tagger.withTagContext(tagContext)) {
+        statsRecorder.newMeasureMap().put(latency, 50).record();
+        statsRecorder.newMeasureMap().put(latency2, 60).record();
+      }
+      // Slow down for OpenCensus to catch up.
+      List<List<MetricData>> result = exporter.waitForNumberOfExports(3);
+      // Just look at last export.
+      List<MetricData> metricData =
+          result.get(2).stream()
+              .sorted(Comparator.comparing(MetricData::getName))
+              .collect(Collectors.toList());
+      assertThat(metricData.size()).isEqualTo(4);
+
+      MetricData metric = metricData.get(0);
+      MetricAssertions.assertThat(metric)
+          .hasName("double_gauge")
+          .hasDescription("double gauge")
+          .hasUnit("ms")
+          .hasDoubleGauge()
+          .points()
+          .satisfiesExactly(
+              point -> MetricAssertions.assertThat(point).hasValue(60).attributes().hasSize(0));
+      metric = metricData.get(1);
+      MetricAssertions.assertThat(metric)
+          .hasName("double_sum")
+          .hasDescription("double sum")
+          .hasUnit("ms")
+          .hasDoubleSum()
+          .points()
+          .satisfiesExactly(
+              point -> MetricAssertions.assertThat(point).hasValue(60).attributes().hasSize(0));
+      metric = metricData.get(2);
+      MetricAssertions.assertThat(metric)
+          .hasName("long_gauge")
+          .hasDescription("long gauge")
+          .hasUnit("ms")
+          .hasLongGauge()
+          .points()
+          .satisfiesExactly(
+              point ->
+                  MetricAssertions.assertThat(point)
+                      .hasValue(50)
+                      .attributes()
+                      .hasSize(1)
+                      .containsEntry(tagKey.getName(), tagValue.asString()));
+      metric = metricData.get(3);
+      MetricAssertions.assertThat(metric)
+          .hasName("long_sum")
+          .hasDescription("long sum")
+          .hasUnit("ms")
+          .hasLongSum()
+          .points()
+          .satisfiesExactly(
+              point ->
+                  MetricAssertions.assertThat(point)
+                      .hasValue(50)
+                      .attributes()
+                      .hasSize(1)
+                      .containsEntry(tagKey.getName(), tagValue.asString()));
+    } finally {
+      otelExporter.stop();
+    }
+  }
+
+  // Straight copy-paste from PeriodicMetricReaderTest.  Should likely move into metrics-testing.
```

Comment:
I think our standard pattern can be InMemory + awaitility - any reason this class would also be needed?

Reply:
No, just need to re-work some numbers, but that's fine by me. I actually couldn't remember the name of the await utiltiy....